src/octopus-cli-wrapper.ts

```diff
@@ -152,13 +162,22 @@ export class OctopusCliWrapper {
     return { args: launchArgs, env: launchEnv }
   }
 
+  async createBuildSummary(): Promise<void> {
+    if (this.pushedPackages.length > 0) {
+      await summary
+        .addHeading(`🎉 Package${this.pushedPackages.length > 1 ? 's' : ''} successfully pushed to Octopus Deploy`, 3)
+        .addCodeBlock(this.pushedPackages.map(pkg => `📦 ${pkg}`).join('\n'))
+        .write()
+    }
+  }
+
   async pushPackage(): Promise<void> {
     info('🔣 Parsing inputs...')
     const cliLaunchConfiguration = this.generateLaunchConfig()
 
     const options: ExecOptions = {
       listeners: {
-        stdline: input => this.stdline(input)
+        stdline: async input => await this.stdline(input)
```

Comment:
ExecOptions.stdline expects a void function, so it will ignore the Promise here and won't await it.

In practice it's probably OK, but worst-case scenario is that perhaps lines could get written out of order because it will call stdline in quick succession while previous async tasks are floating in the background?

The async part seems to be because you used the GitHub actions markdown summary builder thing.
Suggest keeping stdline as a sync function, and just collect all the stuff into an array, and call createBuildSummary later on after the process has completed.